plugins/modules/ipadnsforwardzone.py

```diff
@@ -0,0 +1,246 @@
+            state=dict(type='str', aliases=['ensure'], default='present',
+                       choices=['present', 'absent']),
+        ),
+        supports_check_mode=True,
+    )
+
+    ansible_module._ansible_debug = True
+
+    # Get parameters
+
+    # general
+    ipaadmin_principal = module_params_get(ansible_module,
+                                           "ipaadmin_principal")
+    ipaadmin_password = module_params_get(ansible_module,
+                                          "ipaadmin_password")
+    name = module_params_get(ansible_module, "name")
+    action = module_params_get(ansible_module, "action")
+
+    # present
+    forwarders = module_params_get(ansible_module, "forwarders")
+    forwardpolicy = module_params_get(ansible_module, "forwardpolicy")
+    skip_overlap_check = module_params_get(ansible_module,
+                                           "skip_overlap_check")
+
+    # state
+    state = module_params_get(ansible_module, "state")
+
+    # Check parameters
+    if state == "absent":
+        invalid = ["forwarders", "forwardpolicy", "skip_overlap_check"]
+        for x in invalid:
+            if vars()[x] is not None:
+                ansible_module.fail_json(
+                    msg="Argument '%s' can not be used with action "
+                    "'%s'" % (x, action))
+
+    if state == "present":
+        required = ["forwarders"]
+        for x in required:
+            if vars()[x] is None:
+                ansible_module.fail_json(
+                    msg="Argument '%s' is required when state=present "
+                    "'%s'" % (x, action))
+
+    # Init
+
+    changed = False
+    exit_args = {}
+    args = {}
+    ccache_dir = None
+    ccache_name = None
+    try:
+        if not valid_creds(ansible_module, ipaadmin_principal):
+            ccache_dir, ccache_name = temp_kinit(ipaadmin_principal,
+                                                 ipaadmin_password)
+        api_connect()
+
+        # Make sure forwardzone exists
+        res_find = find_dnsforwardzone(ansible_module, name)
+        if action == "member":
+            forwarders = list(set(res_find["idnsforwarders"] + forwarders))
+
+        try:
+            if res_find is not None:
+                # record exists
+                if state == "present":
+                    # exists and should, has it changed?
+                    args = gen_args(forwarders, forwardpolicy,
+                                    skip_overlap_check)
+                    if skip_overlap_check is not None:
+                        del args['skip_overlap_check']
+
+                    if not compare_args_ipa(ansible_module, args, res_find):
+                        changed = True
+                        result = api_command(ansible_module,
+                                             "dnsforwardzone_mod",
+                                             name, args)
+                elif state == "absent":
+                    # exists but shouldn't
```

Comment:
Why shouldn't the entry exist? With state: absent an existing entry should get removed.